apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LoggingConfiguration.java

```diff
@@ -117,67 +122,94 @@ public void assertValid(Boolean value) {
         })
         .buildWithDefault(false);
 
-    public static void init(List<ConfigurationSource> sources) {
-        setLogLevel(getValue(LOG_LEVEL_KEY, sources,
-            getValue(DEPRECATED_LOG_LEVEL_KEY, sources, LogLevel.INFO.toString())));
-        setLogFileLocation(ElasticApmAgent.getAgentHome(), getValue(LOG_FILE_KEY, sources,
-            getValue(DEPRECATED_LOG_FILE_KEY, sources, DEFAULT_LOG_FILE)));
-    }
+    @SuppressWarnings("unused")
+    public ConfigurationOption<ByteValue> logFileMaxSize = ByteValueConverter.byteOption()
+        .key(LOG_FILE_MAX_SIZE_KEY)
+        .configurationCategory(LOGGING_CATEGORY)
+        .description("The max size of the log file.\n" +
+            "\n" +
+            //"To support <<config-ship-agent-logs,shipping the logs>> to APM Server,\n" +
+            "The agent always keeps one history file so that the max total log file size is twice the value of this setting.\n")
```

Comment:
> there is some commented code, probably leftovers

This is intentionally left in as it will be needed when we support shipping logs.

> I feel that we could make this less confusing by using a value that is half of the configured value.

The reason why I did it this way is that `log_file_max_size` also determines how much is buffered on disk before data is deleted. When shipping logs, it basically determines how much data can be buffered without data loss while the APM Server is down. So they might expect that we always buffer at least `log_file_max_size`. We can only guarantee that, if both the primary and the history log file have the max size of `log_file_max_size`.

Reply:
Then, probably removing the 'max' from the configuration would remove any expectation to be the actual max disk usage by the agent. for example if we name it `log_file_buffer_size` and then document that disk usage is at most equal to twice this value.